File: next.py
from problog.extern import problog_export
times = []

@problog_export('+list','-int')
def collect_all(a):
    times.append(a)
    return 1

@problog_export('-list')
def sort():
    return [Tuple((Number(i), val)) for i, val in enumerate(sorted(times))]

File: next.pl
:- use_module('next.py').
collect_all.
collect_all :- time(X), collect_all(X) == 0.
sorted_pair(X,N) :- collect_all, (X,N) = sort().
next(X, Y) :- sorted_pair(A,X), sorted_pair(A+1,Y).

